tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -799,6 +799,68 @@ public:
    void CommitCluster() final;
 };
 
+/// The field for values that may or may not be present in an entry. Parent class for unique pointer field and
+/// optional field. A nullable field cannot be instantiated itself but only its descendants.
+/// The RNullableField takes care of the on-disk representation. Child classes are responsible for the in-memory
+/// representation.  The on-disk representation can be "dense" or "sparse". Dense nullable fields have a bitmask
+/// (true: item available, false: item missing) and serialize a default-constructed item for missing items.
```

Comment:
If we already know the items is not available do we really need to serialize an default item?

Reply:
That is true but it would make reading more complicated. If we skipped serialization of NULL elements, upon reading of the $k$'s entry in a cluster, we need to count the one bits up to $k$ in order to determine the indexes of the child field columns. At this point, I don't think there would be a benefit of the bitmap anymore over the standard collection representation.

Reply:
Is the default constructed object stored **only** in the sparse case (when is a bit) or in both cases?